Source/Core/VideoCommon/HiresTextures_DDSLoader.cpp

```diff
@@ -165,29 +142,95 @@ struct DDSLoadInfo
   size_t first_mip_offset = 0;
   size_t first_mip_size = 0;
   u32 first_mip_row_length = 0;
+
+  std::function<void(HiresTexture::Level*)> conversion_function;
 };
 
-static u32 GetBlockCount(u32 extent, u32 block_size)
+u32 GetBlockCount(u32 extent, u32 block_size)
 {
   return std::max(Common::AlignUp(extent, block_size) / block_size, 1u);
 }
 
-static u32 CalculateMipCount(u32 width, u32 height)
+HiresTexture::ImageDataPointer AllocateLevelData(size_t size)
+{
+  return HiresTexture::ImageDataPointer(new u8[size], [](u8* data) { delete[] data; });
+}
+
+void ConvertTexture_X8B8G8R8(HiresTexture::Level* level)
 {
-  u32 mip_width = std::max(width / 2, 1u);
-  u32 mip_height = std::max(height / 2, 1u);
-  u32 mip_count = 1;
-  while (mip_width > 1 || mip_height > 1)
+  u8* data_ptr = level->data.get();
+  for (u32 row = 0; row < level->height; row++)
   {
-    mip_width = std::max(mip_width / 2, 1u);
-    mip_height = std::max(mip_height / 2, 1u);
-    mip_count++;
+    for (u32 x = 0; x < level->row_length; x++)
```

Comment:
Shouldn't produce any different code, as far as I'm aware. Only exception would be for iterators, but most compilers should be smart enough to optimize away the saving-of-current-value for post-increment.

IMO, pre-increment is clearer than post-increment.